lib/src/repo_path.rs

```diff
@@ -502,6 +503,70 @@ impl RepoPathUiConverter {
         }
     }
 
+    /// Format a copy from `source` to `target` for display in the UI by
+    /// extracting common components and producing something like
+    /// "common/prefix/{source => target}/common/suffix".
+    ///
+    /// If `source == target`, returns `format_file_path(source)`.
```

Comment:
I'm not sure this belongs in the lib crate. It feels a bit specific to a CLI. It seems likely that e.g. a GUI or Web UI would want to display it differently. Actually, even the CLI may want to style the renamed part differently from the rest of the string. So I feel like this should be made more generic or be moved into CLI. It could be made more generic by returning the `(prefix, (before, after), suffix)` or something like that. Maybe a proper struct or enum is better (an enum could have a `Unchanged(String)` variant, but it could instead be a struct wrapped in an `Option`). What do you think?

Reply:
do you have a suggestion for where to move it?  One of the reasons it fits here in my mind is that `format_file_path` is also doing a bunch of UI specific work and they do share behavior that core part of things.

Reply:
I think `format_file_path()` is generic enough that it makes sense in a web UI too, for example, but a web UI would probably not want to display the `common/prefix/{source => target}/common/suffix` form. Maybe `diff_util.rs` is a good place since this way of displaying paths is specific to diffs.

Reply:
Oh, but maybe the use of `RepoPathUiConverter` means that it should be here. Let me take a look at the code to see how this works.

Reply:
I'm still not sure where this code should live but we can always move it later. Let's leave it here for now unless someone else has a good idea for how to structure it. Sorry about my indecision.